### PR TITLE
Comment Widget

### PR DESCRIPTION
**Pull request by BrandonSharratt**
*Issue:*https://github.com/bcgov/entity/issues/21305

*Description of changes:*
Added a comment widget and hooked into the needed places. Added the ability to use a staff account to cypress tests and used the staff account to test the comment widget.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-26-ujfdt5l2.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-26-ujfdt5l2.web.app

File: cypress/fixtures/settings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [
   {
       "accountStatus": "ACTIVE",
```
Comment (patrickpeinanw) [inline cypress/fixtures/settings.json:3]: Can you change this accountType back to 'PREMIUM' so we don't always use a staff account?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-26-ujfdt5l2.web.app

File: src/components/bcros/comment/Index.vue [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    <slot name="comment-add-area">
+      <div class="mb-6" data-cy="comment-add-slot">
+        <UTextarea
+          v-model="commentToAdd"
+          data-cy="comment-add-textarea"
+          :variant="error ? 'bottomError' : 'bottom'"
+          :rows="5"
+          :ui="textAreaUi"
+        />
+        <p v-if="error" class="text-red-600">
```
Comment (kialj876) [inline src/components/bcros/comment/Index.vue:97]: if you wrap the UTextarea with a UFormGroup then you can remove all these <p> blocks and the styling for text colour will come from the our common file, which would be better I think. You would have to change the 'charactersExceeded' message to come from your error message logic

`<UFormGroup :error="error" :help="MAX_COMMENT_LENGTH - commentToAdd.length">...`

File: src/components/bcros/comment/Index.vue [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          >
+            <span>{{ $t('label.comments.cancel') }}</span>
+          </UButton>
+        </div>
+      </div>
+    </slot>
+
+    <slot name="comment-list">
+      <div
+        class="pt-8 overflow-y-scroll absolute inset-x-0 bottom-5 h-[calc(100%-275px)]"
```
Comment (kialj876) [inline src/components/bcros/comment/Index.vue:130]: can you add px-5 here so that the comments are in line with the text area above?

**Review by kialj876 — COMMENTED** [review]
Looking good, last couple things below. Also I noticed that the *first time* you select 'add detail' on a filing with exiting comments it does not list the previous comments

File: src/components/bcros/comment/Index.vue [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+const MAX_COMMENT_LENGTH = 2000
+const commentToAdd = ref('')
+
+watch(commentToAdd, (newComment, oldComment) => {
+  if (newComment !== oldComment) {
+    noChangesSinceSave.value = false
+  }
+})
+const error = computed((): string => {
```
[the remark this answers is not in the document]
Comment (BrandonSharratt) [reply src/components/bcros/comment/Index.vue]: yea moved all the logic to the watch now which fixed it

File: src/components/bcros/comment/Index.vue [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  if (!commentToAdd || !commentToAdd.value || commentToAdd.value.length === 0) {
+    if (noChangesSinceSave.value === true) {
+      return ''
+    }
+    return t('label.comments.commentRequired')
+  }
+  // if (commentToAdd && commentToAdd.value && commentToAdd.value.length > MAX_COMMENT_LENGTH) {
```
[the remark this answers is not in the document]
Comment (BrandonSharratt) [reply src/components/bcros/comment/Index.vue]: in moving this all to the watch I removed this

File: src/components/bcros/comment/Index.vue (hunk 3 above)
Comment (BrandonSharratt) [reply src/components/bcros/comment/Index.vue:130]: I had previously added it below but I'll move it here

File: cypress/fixtures/settings.json (hunk 1 above)
Comment (BrandonSharratt) [reply cypress/fixtures/settings.json]: good catch done

File: src/components/bcros/comment/Index.vue (hunk 2 above)
Comment (BrandonSharratt) [reply src/components/bcros/comment/Index.vue]: done

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-26-ujfdt5l2.web.app

**Review by kialj876 — APPROVED** [review]
Nice work, thanks for the updates

**Pull request merged**